src/libFLAC/stream_decoder.c

```diff
@@ -2123,13 +2123,15 @@ FLAC__bool read_frame_(FLAC__StreamDecoder *decoder, FLAC__bool *got_a_frame, FL
 		}
 	}
 	else {
-		/* Bad frame, emit error and zero the output signal */
+		/* Bad frame, emit error, zero the output signal, return */
 		send_error_to_client_(decoder, FLAC__STREAM_DECODER_ERROR_STATUS_FRAME_CRC_MISMATCH);
 		if(do_full_decode) {
 			for(channel = 0; channel < decoder->private_->frame.header.channels; channel++) {
 				memset(decoder->private_->output[channel], 0, sizeof(FLAC__int32) * decoder->private_->frame.header.blocksize);
 			}
 		}
+		decoder->protected_->state = FLAC__STREAM_DECODER_SEARCH_FOR_FRAME_SYNC;
+		return true;
```

Comment:
Can you clarify (perhaps in commit message) why it should return true? Some of the other early exits return false. It doesn't seem obvious that the existing code is consistent and correct, so it is non-trivial to argue that this is correct.